torch/csrc/jit/ir/alias_analysis.cpp

```diff
@@ -12,25 +12,51 @@ namespace jit {
 
 namespace {
 
-// For any mutable type, map it to a type such that all other types which it can
-// alias will be mapped to the same type. This function follows a similar logic
-// to `unifyTypes` because any two mutable types which can be unified
-// can alias each other.
-// getMutableTypePtr(Optional[List[int]]) == getMutableTypePtr([List[int]])
-// If a type is not mutable, return nullopt
-// This class helps convert types to their mutable equivalent by looking up
-// cached conversions.
+// This class determines whether a type is mutable, and, if so, it maps
+// the type to its "mutable equivalent" (see definition in
+// `getMutableType`). It uses a cache of TypePtrs to speed up these
+// type lookups
 class MutableTypePtrHelper {
  public:
   explicit MutableTypePtrHelper(
       std::unordered_map<TypePtr, TypePtr>* mutable_type_cache)
       : mutable_type_cache_(mutable_type_cache) {}
 
+  // Map any mutable type to a type such that all other types which the
+  // mutable type can alias will be mapped to the same type. This
+  // follows a similar logic to `unifyTypes` because any two mutable
```

Comment:
Same semantics, but imo the original wording could be improved. I had to read that first sentence several times to picture it in my head. I don't mind keeping the original if I'm the only one who has problems with it, though :)

Reply:
The semantics are a bit different, as the `Tuple[None, Tensor]` and `Tuple[List[int], None]` case show

Reply:
Is the issue here with "This function follows a similar logic to `unifyTypes` because..."? That line is from the original wording. It was confusing to me when I was trying to understand the alias analysis pass (and it sounds like you're saying it's downright misleading), so maybe we should take it out altogether